File: rpi/kxr94-2050/src/kxr94_2050_sample.py
import time

from gpiozero import MCP3204


ADC_RESOLUTION = 4096  # 12-bit
REFERENCE_VOLTAGE = 3300  # mV
VOLTAGE_PER_G = REFERENCE_VOLTAGE / 5  # mV
OFFSET_VOLTAGE = REFERENCE_VOLTAGE / 2  # mV
GAL = 980.665  # cm/s^2


def main():
    mcp3204 = [
        MCP3204(channel=0),
        MCP3204(channel=1),
        MCP3204(channel=2),
    ]
    tri_axis = [0, 0, 0]

    while True:
        for i in range(3):
            tri_axis[i] = (
                mcp3204[i].raw_value
                / ADC_RESOLUTION
                * REFERENCE_VOLTAGE
            )

            tri_axis[i] -= OFFSET_VOLTAGE
            tri_axis[i] /= VOLTAGE_PER_G
            tri_axis[i] *= GAL

        print(f'X: {tri_axis[0]} Y: {tri_axis[1]} Z: {tri_axis[2]}')
        time.sleep(0.1)


if __name__ == '__main__':
    main()

File: rpi/kxr94-2050/go/examples/kxr94_2050_sample.go
package main

import (
	"fmt"
	"time"

	"gobot.io/x/gobot"
	"gobot.io/x/gobot/drivers/spi"
	"gobot.io/x/gobot/platforms/raspi"
)

const adcResolution float32 = 4096                 // 12-bit
const referenceVoltage float32 = 3300              // mV
const voltagePerG float32 = referenceVoltage / 5   // mV
const offsetVoltage float32 = referenceVoltage / 2 // mV
const gal float32 = 980.665                        // cm/s^2

func main() {
	r := raspi.NewAdaptor()
	adc := spi.NewMCP3204Driver(r)

	var axis [3]float32

	work := func() {
		gobot.Every(100*time.Millisecond, func() {
			for i := 0; i < 3; i++ {
				result, _ := adc.Read(i)

				axis[i] = float32(result) / adcResolution * referenceVoltage
				axis[i] -= offsetVoltage
				axis[i] /= voltagePerG
				axis[i] *= gal
			}

			fmt.Printf("X: %g Y: %g Z: %g", axis[0], axis[1], axis[2])
		})
	}

	robot := gobot.NewRobot("accelBot",
		[]gobot.Connection{r},
		[]gobot.Device{adc},
		work,
	)

	robot.Start()
}
